packages/commonwealth/client/scripts/state/index.ts

```diff
@@ -192,44 +185,47 @@ export async function initAppState(
   shouldRedraw = true,
 ): Promise<void> {
   try {
-    const [{ data: statusRes }, { data: communities }, { data: nodesRes }] =
-      await Promise.all([
-        axios.get(`${app.serverUrl()}/status`),
-        axios.get(`${app.serverUrl()}/communities`),
-        axios.get(`${app.serverUrl()}/nodes`),
-      ]);
+    const [{ data: statusRes }, { data: communities }] = await Promise.all([
+      axios.get(`${app.serverUrl()}/status`),
+      axios.get(`${app.serverUrl()}/communities`),
+    ]);
+
+    const nodesData = await fetchNodesQuery();
 
     app.config.chains.clear();
-    app.config.nodes.clear();
     app.user.notifications.clear();
     app.user.notifications.clearSubscriptions();
-    app.config.evmTestEnv = statusRes.result.evmTestEnv;
-    app.config.enforceSessionKeys = statusRes.result.enforceSessionKeys;
 
-    nodesRes.result
-      .sort((a, b) => a.id - b.id)
-      .forEach((node) => {
-        app.config.nodes.add(NodeInfo.fromJSON(node));
-      });
+    queryClient.setQueryData([QueryKeys.CONFIGURATION], {
+      enforceSessionKeys: statusRes.result.enforceSessionKeys,
+      evmTestEnv: statusRes.result.evmTestEnv,
+    });
 
     communities.result
       .filter((c) => c.community.active)
       .forEach((c) => {
         const chainInfo = ChainInfo.fromJSON({
-          ChainNode: app.config.nodes.getById(c.community.chain_node_id),
+          ChainNode: nodesData.find((n) => n.id === c.community.chain_node_id),
           ...c.community,
         });
         app.config.chains.add(chainInfo);
+
         if (chainInfo.redirect) {
-          app.config.redirects[chainInfo.redirect] = chainInfo.id;
+          const cachedConfig = queryClient.getQueryData<Configuration>([
+            QueryKeys.CONFIGURATION,
+          ]);
+
+          queryClient.setQueryData([QueryKeys.CONFIGURATION], {
```

Comment:
We should check with product if we can get rid of the redircts. We have so much custom code, and it only redirects redacted-cartel -> redacted-protocol

Reply:
This is (almost) definitely a feature we should keep. We have redirects for: 'common', 'redacted-protocol', 'picasso'. If we remove it we will most likely need to add it back in the future.

Reply:
common doesn't work: https://commonwealth.im/common/discussions
picaso doesn't work: https://commonwealth.im/picasso/discussions

My problem with this feature is that it is a hacky workaround. Maybe we could have a recently renamed community section on the homepage for visibility, but to actually change the logic of the app makes things confusing.

For example what happens if there already exists a community with an id with the redirected url? What community would the url go to?

Reply:
> My problem with this feature is that it is a hacky workaround. Maybe we could have a recently renamed community section on the homepage for users that have joined that community for visibility. Additionally we could allow admins to close a community, and allow them to display a redirect screen on that closed community that points to a new community. But this should be solved with our existing tools, not some custom routing functionality

I disagree. It's not a 'workaround' it's a feature. What you are describing are completely different features, not solutions to the same problem. Your solutions do not account for the fact that an old link to a thread would become useless and no longer point to the same relevant thread on a renamed community.

> For example what happens if there already exists a community with an id with the redirected url (or one is created after the fact)? What community would the url go to?

Redirected id's are reserved. You can't create a community if there is already a redirect id for it and vice versa for an existing community.

Reply:
I was just thinking more of like if we just made them create a new community while deprecating their old community and having a redirect thread, then it would not run into the following issues:
* The SEO will get destroyed for old pages on redirected communities because google doesn't like when it auto-redirects to a new page.
* The front end routing logic is disjoint. We have the majority of our routing logic at the startup of the app, then we have this additional routing logic in the Layout page
* Extra overhead on the backend, now we need to ensure all communities are loaded at startup because we need every communities redirect list (Although we have is this way currently with the /status anyways).

Anyways not a huge deal to keep it the way things are.

Reply:
thanks for this conversation. Because it currently touches 2 or 3 communities, I think we sill stick what we have right now. But I see your point that it could be improved in some ways. We will keep that in mind

Reply:
> I was just thinking more of like if we just made them create a new community while deprecating their old community and having a redirect thread, then it would not run into the following issues:

> * The SEO will get destroyed for old pages on redirected communities because google doesn't like when it auto-redirects to a new page.

Where did you get this? JS redirects are generally not recommended because a crawler won't necessarily wait for the redirect to render. This is not a problem for us because we use prerender.

> * Extra overhead on the backend, now we need to ensure all communities are loaded at startup because we need every communities redirect list (Although we have is this way currently with the /status anyways, but we will hopefully migrate away from it).

Loading all communities is not technically required. You could easily search over redirect community IDs if a primary community ID is not found...